docs/use_cases/otel/preprocessing.md

```diff
@@ -0,0 +1,153 @@
+# Ingress
+
+To push data into Feldera, we configure the OpenTelemetry (OTel) Collector to send data to Feldera via HTTP.
+
+#### OTel Collector Configuration:
+
+```yml
+exporters:
+
+  # Send traces to the Feldera pipeline.
+  # Currently only uncompressed JSON is supported.
+  otlphttp/feldera_traces:
+     traces_endpoint: http://feldera:8080/v0/pipelines/otel/ingress/otel_traces?format=json&update_format=raw
+     encoding: json
+     compression: none
+
+  otlphttp/feldera_logs:
+     logs_endpoint: http://feldera:8080/v0/pipelines/otel/ingress/otel_logs?format=json&update_format=raw
+     encoding: json
+     compression: none
+
+  otlphttp/feldera_metrics:
+     metrics_endpoint: http://feldera:8080/v0/pipelines/otel/ingress/otel_metrics?format=json&update_format=raw
+     encoding: json
+     compression: none
+
+service:
+   pipelines:
+     traces:
+       exporters: [spanmetrics, otlphttp/feldera_traces]
+
+     logs:
+       receivers: [otlp]
+       exporters: [otlphttp/feldera_logs]
+
+     metrics:
+       receivers: [hostmetrics, docker_stats, httpcheck/frontend-proxy, otlp, prometheus, redis, spanmetrics]
+       exporters: [otlphttp/feldera_metrics]
+```
+
+Notice that we have the query params: `format=json&update_format=raw`
+- `format=json`: Specifies that the input data is JSON-formatted.
+- `update_format=raw`: Indicates that the request body contains raw data, which is processed as an insertion into the table.
+
+## Preprocessing
+
+Next we perform a series of `UNNEST` operations to extract individual rows of metrics, logs and traces from a nested arrays of them.
+
+#### Step 1: Extract Resource-Level Data
+
+```sql
+-- (ResouceMetrics[N]) -> (Resource, ScopeMetrics[N])
+CREATE LOCAL VIEW rsMetrics AS SELECT resource, scopeMetrics
+FROM otel_metrics, UNNEST(resourceMetrics) as t (resource, scopeMetrics);
+
+-- (ResouceSpans[N]) -> (Resource, ScopeSpans[N])
+CREATE LOCAL VIEW rsSpans AS SELECT resource, scopeSpans
+FROM otel_traces, UNNEST(resourceSpans) as t (resource, scopeSpans);
+
+-- (ResouceLogs[N]) -> (Resource, ScopeLogs[N])
+CREATE LOCAL VIEW rsLogs AS SELECT resource, scopeLogs
+FROM otel_logs, UNNEST(resourceLogs) as t (resource, scopeLogs);
+```
+
+Here we `UNNEST`  arrays of `ResourceSpans`, `ResourceMetrics` and `ResourceLogs` into separate rows.
```

Comment:
Some observability people may not be familiar with SQL, so it may pay off to have a simple running example with a concrete event that shows how it will look in each view.